modules/oe_content_person/tests/src/FunctionalJavascript/PersonEntityTest.php

```diff
@@ -136,18 +136,43 @@ public function testPersonForm() {
     $this->assertTrue($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_reference][0][target_id]')->isVisible());
     $this->assertTrue($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_acting][value]')->isVisible());
     $this->assertTrue($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_description][0][value]')->isVisible());
-    $this->assertFalse($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_name][0][value]')->isVisible());
-    $this->assertFalse($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_name][0][value]')->hasAttribute('required'));
-    $this->assertEquals('required', $this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_reference][0][target_id]')->getAttribute('required'));
+    $this->assertTrue($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_name][0][value]')->isVisible());
 
     // Assert the job does not have a reference role.
     $this->assertEmpty($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_reference][0][target_id]')->getValue());
+    // But it has the custom role.
+    $this->assertSession()->fieldValueEquals('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_name][0][value]', 'Custom Role');
+
+    // Save and assert the person is updated.
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->assertSession()->pageTextContains('Person John Doe has been updated.');
+
+    // Edit and assert the values were kept.
+    $this->drupalGet('/node/1/edit');
+    $job_region = $this->getSession()->getPage()->find('css', '#edit-oe-person-jobs-wrapper');
+    $job_region->pressButton('Edit');
+    $this->assertSession()->assertWaitOnAjaxRequest();
+    $this->assertEmpty($this->getSession()->getPage()->findField('oe_person_jobs[form][inline_entity_form][entities][0][form][oe_role_reference][0][target_id]')->getValue());
+    file_put_contents('form.html', $this->getSession()->getPage()->getOuterHtml());
```

Comment:
A leftover 

BTW, it's possible to output all HTML from all `->drupalGet()` calls, like in standard Drupal tests. 

First, set these env vars in docker-compose overrides: 
```shell
      BROWSERTEST_OUTPUT_DIRECTORY: /var/www/html/some_existing_dir
      BROWSERTEST_OUTPUT_BASE_URL: "http://localhost:8080/build"
```
And then run phpunit like this
```shell
docker compose exec web vendor/bin/phpunit --printer=\\Drupal\\Tests\\Listeners\\HtmlOutputPrinter
```
I found this one recently, it was turned off because it generates tons of HTML 😉.

Reply:
Thanks!